content-src/selectors/selectors.js

```diff
@@ -1,144 +1,68 @@
+ *    .TopActivity {obj} State object (empty for weightedHighlights experiment)
+ *    .isReady {bool} Do we have all the required data to display New Tab?
+ *    .showRecommendationOption {bool} Should we show the option to turn recommendations off/on?
  */
-const selectAndDedupe = module.exports.selectAndDedupe = options => {
-  let rows = dedupe.group([
-    options.dedupe,
-    options.sites])[1];
-
-  if (rows.length < options.max && options.defaults) {
-    rows = rows.concat(options.defaults);
-  }
-
-  return rows.slice(0, options.max);
-};
-
-const selectWeightedHighlights = module.exports.selectWeightedHighlights = createSelector(
-  [
-    state => state.WeightedHighlights
-  ],
-  WeightedHighlights => Object.assign({}, WeightedHighlights, {rows: assignImageAndBackgroundColor(WeightedHighlights.rows)})
-);
-
-const selectSpotlight = module.exports.selectSpotlight = createSelector(
-  [
-    state => state.Highlights,
-    state => state.Prefs.prefs.recommendations,
-    selectWeightedHighlights,
-    state => state.Experiments.values.weightedHighlights
-  ],
-  (Highlights, recommendationShown, WeightedHighlights, prefWeightedHighlights) => {
-    let rows;
-    if (prefWeightedHighlights) {
-      rows = WeightedHighlights.rows;
-    } else {
-      // Only concat first run data if init is true
-      const highlightRows = Highlights.rows.concat(Highlights.init ? firstRunData.Highlights : []);
-      rows = assignImageAndBackgroundColor(highlightRows)
-        .sort((site1, site2) => {
-          const site1Valid = isValidSpotlightSite(site1);
-          const site2Valid = isValidSpotlightSite(site2);
-          if (site2.type === firstRunData.FIRST_RUN_TYPE) {
-            return -1;
-          } else if (site1.type === firstRunData.FIRST_RUN_TYPE) {
-            return 1;
-          } else if (site1Valid && site2Valid) {
-            return 0;
-          } else if (site2Valid) {
-            return 1;
-          }
-          return -1;
-        });
-    }
-
-    return Object.assign({}, Highlights, {rows, recommendationShown, prefWeightedHighlights});
-  }
-);
-
-const selectTopSites = module.exports.selectTopSites = createSelector(
-  [
-    state => state.TopSites
-  ],
-  TopSites => Object.assign({}, TopSites, {
-    rows: dedupe.one(TopSites.rows
-      // Add first run stuff to the end if init has already happened
-      .concat(TopSites.init ? firstRunData.TopSites : []))
-  })
-);
-
 module.exports.selectNewTabSites = createSelector(
   [
-    selectWeightedHighlights,
     state => state.Experiments.values.weightedHighlights,
-    selectTopSites,
-    state => state.History,
-    selectSpotlight,
+    state => (state.Experiments.values.weightedHighlights ? state.WeightedHighlights : selectSpotlight(state)),
+    state => state.TopSites,
+    state => (state.Experiments.values.weightedHighlights ? Object.assign({}, state.History, {rows: []}) : state.History),
```

Comment:
just to be sure i understand: this is saying if the weighted highlights are on, we don't want to even populate the history selector (aka recent items)